### ISSUE TITLE
DankNull HUD missing when graphics set to "Fast" preset

### ISSUE THREAD

**Issue by chimericdream** [p455w0rd/DankNull#64]
In one of the latest updates, I noticed that the HUD overlay was no longer showing up when I am holding the DankNull in my hand. Was this a deliberate reversion, or did it move to an optional config somewhere that I missed?

MC: 1.12.2
DN: 1.12-1.3.29

Thanks!

**Comment by p455w0rd** [comment]
it doesnt show if it's empty ..is this the case?

**Comment by chimericdream** [comment]
Unfortunately not. I'll try emptying it entirely and putting stuff back. If that doesn't work, I'll cheat in the materials to make a new one from scratch and report back.

Could there be something funky in the NBT data for the DN since I've upgraded the mod several times since I crafted it?

**Comment by p455w0rd** [comment]
no..metadata is used to determine the level...gotta be another mod cancelling the overlay event

**Comment by chimericdream** [comment]
I'll try to dig a little further then. I just got home from work and haven't logged in to test yet. Do you know if there's a way to identify the culprit? I haven't added any mods in a little while, though I have updated quite a few. Worst case, I can clone the profile and go the trial and error route.

**Comment by p455w0rd** [comment]
well if you want, just create a completely new profile and just install the latest 1.12.2 /dank/null..it'll grab the necessary deps...if it still doesn't show then it's 100% on my end

**Comment by chimericdream** [comment]
Can confirm. A clean profile with the following setup had no HUD overlay.

Minecraft 1.12.2
Forge 14.23.1.2566
/dank/null 1.12-1.3.29
CodeChickenLib 1.12.2-3.1.5.331-universal
p455w0rd's Library 1.12-2.0.24

**Comment by p455w0rd** [comment]
ty! will look into it first thing tomorrow..seriously..you have been such a big help...I truly appreciate it!

**Comment by chimericdream** [comment]
Not a problem. As a developer myself, I try to submit bug reports for issues in mods if/when I come across them. And I _really_ appreciate your crazy fast response time!

**Comment by p455w0rd** [comment]
lol..i owe it all to webhooks..i get a message in my discord "#git" channel every time one of my mods gets a report XD

**Comment by p455w0rd** [comment]
@chimericdream is the /dank/null in your off hand by chance when this happens? I noticed this myself and that much is fixed in dev..beyond that, I'm baffled

**Comment by chimericdream** [comment]
There's no HUD in off hand or primary hand. With any luck, whatever change you made to fix the off hand issue will fix it overall. Out of curiosity, did the same clean setup work as expected for you, or did the error reproduce itself? From your message, it sounds like the issue isn't present for you with the same setup.

**Comment by p455w0rd** [comment]
only diff would be pwlib..I'm using 2.0.25

**Comment by chimericdream** [comment]
I just confirmed that with the setup below, the HUD is visible again. I'll double check my pack for version differences later today and report back. Could be that whatever issue I was seeing is fixed on your end but is still being interrupted by another mod.

Minecraft 1.12.2
Forge 14.23.1.2566 (and 2584)
/dank/null 1.12-1.3.30
CodeChickenLib 1.12.2-3.1.5.331-universal
p455w0rd's Library 1.12-2.0.25

**Comment by chimericdream** [comment]
I FIGURED IT OUT. It's not a mod compatibility thing at all; instead, it appears that if you change the video settings from "Fancy" to "Fast", the HUD disappears. Confirmed with the following profile:

Forge: 14.23.2.2615
/dank/null: 1.4.31
CodeChicken Lib 1.8.+: 3.1.5.331
p455w0rd's Library: 2.0.28

Just to be sure, I toggled the setting back and forth a few times. Whenever I was on "Fancy" graphics, the HUD was present, and it disappeared on the "Fast" graphics. I honestly have no idea why this would matter, but hopefully this gives you more to go on (at least it's reproducible now).

I can haz reopened bug report?

:grin:
